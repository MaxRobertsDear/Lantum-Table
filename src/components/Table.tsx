import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import Paper from '@material-ui/core/Paper'
import Table from '@material-ui/core/Table'
import TableBody from '@material-ui/core/TableBody'
import TableCell from '@material-ui/core/TableCell'
import TableContainer from '@material-ui/core/TableContainer'
import TableHead from '@material-ui/core/TableHead'
import TablePagination from '@material-ui/core/TablePagination'
import TableRow from '@material-ui/core/TableRow'

const useStyles = makeStyles({
  root: {
    width: '100%',
  },
  container: {
    maxHeight: 440,
  },
})

const ResuableTable = ({ rows, columns }: any) => {
  const classes = useStyles()
  const [page, setPage] = React.useState(0)
  const [rowsPerPage, setRowsPerPage] = React.useState(10)

  const handleChangePage = (event: unknown, newPage: number) => {
    setPage(newPage)
  }

  const handleChangeRowsPerPage = (
    event: React.ChangeEvent<HTMLInputElement>,
  ) => {
    setRowsPerPage(+event.target.value)
    setPage(0)
  }

  return (
    <Paper className={classes.root}>
      <TableContainer className={classes.container}>
        <Table stickyHeader aria-label='sticky table'>
          <TableHead>
            <TableRow>
              {columns &&
                columns.map(
                  (column: {
                    id: string,
                    align:
                      | 'left'
                      | 'right'
                      | 'inherit'
                      | 'center'
                      | 'justify'
                      | undefined,
                    minWidth: any,
                    label: React.ReactNode,
                  }) => (
                    <TableCell
                      key={column.id}
                      align={column.align}
                      style={{ minWidth: column.minWidth }}
                    >
                      {column.label}
                    </TableCell>
                  ),
                )}
            </TableRow>
          </TableHead>
          <TableBody>
            {rows &&
              rows
                .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
                .map(
                  (row: {
                    [x: string]: any,
                    id: string | number | null | undefined,
                  }) => {
                    return (
                      <TableRow
                        hover
                        role='checkbox'
                        tabIndex={-1}
                        key={row.id}
                      >
                        {columns.map(
                          (column: {
                            id: string,
                            align:
                              | 'left'
                              | 'right'
                              | 'inherit'
                              | 'center'
                              | 'justify'
                              | undefined,
                          }) => {
                            const value = row[column.id]
                            return (
                              <TableCell key={column.id} align={column.align}>
                                {value}
                              </TableCell>
                            )
                          },
                        )}
                      </TableRow>
                    )
                  },
                )}
          </TableBody>
        </Table>
      </TableContainer>
      <TablePagination
        rowsPerPageOptions={[10, 25, 100]}
        component='div'
        count={rows.length}
        rowsPerPage={rowsPerPage}
        page={page}
        onChangePage={handleChangePage}
        onChangeRowsPerPage={handleChangeRowsPerPage}
      />
    </Paper>
  )
}
export default ResuableTable
