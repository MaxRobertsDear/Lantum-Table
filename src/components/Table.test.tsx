import React from 'react'
import { render } from '@testing-library/react'
import Table from './Table'
import { columns, rows } from './mock.data'

test('renders Table', async () => {
  const { container } = render(<Table rows={rows} columns={columns} />)
  expect(container).toMatchSnapshot()
})
