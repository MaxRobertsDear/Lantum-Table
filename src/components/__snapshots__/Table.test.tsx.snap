// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Table 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiTableContainer-root makeStyles-container-2"
    >
      <table
        aria-label="sticky table"
        class="MuiTable-root MuiTable-stickyHeader"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader"
              scope="col"
              style="min-width: 170px;"
            >
              Practice
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader"
              scope="col"
              style="min-width: 100px;"
            >
              Date
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-stickyHeader"
              scope="col"
              style="min-width: 170px;"
            >
              Start and End Time
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
              scope="col"
              style="min-width: 170px;"
            >
              Hourly Rate
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
              scope="col"
              style="min-width: 170px;"
            >
              Number of applicants
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-hover"
            role="checkbox"
            tabindex="-1"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              Grant Tree Medical Centre
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              19/11/2020
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
            >
              19/11/2020, 16:50:00 - 19/11/2020, 19:15:00
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              85
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            />
          </tr>
          <tr
            class="MuiTableRow-root MuiTableRow-hover"
            role="checkbox"
            tabindex="-1"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              Central London Hospital
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              19/03/2021
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
            >
              19/03/2021, 08:50:00 - 19/03/2021, 17:15:00
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              100
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            />
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="MuiTablePagination-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
      >
        <div
          class="MuiTablePagination-spacer"
        />
        <p
          class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
          id="mui-52237"
        >
          Rows per page:
        </p>
        <div
          class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby="mui-52237 mui-55411"
            class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
            id="mui-55411"
            role="button"
            tabindex="0"
          >
            10
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput"
            tabindex="-1"
            value="10"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
        <p
          class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
        >
          1-2 of 2
        </p>
        <div
          class="MuiTablePagination-actions"
        >
          <button
            aria-label="Previous page"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            title="Previous page"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
            </span>
          </button>
          <button
            aria-label="Next page"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            title="Next page"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
